library/spark/spark-accumulo-library/src/main/java/uk/gov/gchq/gaffer/sparkaccumulo/operation/handler/scalardd/GetRDDOfAllElementsHandler.java

```diff
@@ -45,13 +85,139 @@
+            final IteratorSetting itrSetting = accumuloStore
+                    .getKeyPackage().getIteratorFactory().getAggregatorIteratorSetting(accumuloStore);
+            InputConfigurator.addIterator(AccumuloInputFormat.class, conf, itrSetting);
+        } else {
+            LOGGER.info("Not adding aggregator iterator as aggregation is not enabled");
+        }
+    }
+
+    public static class EntryIteratorToElementIterator
+            extends AbstractFunction1<Iterator<Map.Entry<Key, Value>>, Iterator<Element>> implements Serializable {
+        private byte[] serialisedConf;
+
+        public EntryIteratorToElementIterator(final byte[] serialisedConf) {
+            this.serialisedConf = serialisedConf;
+        }
+
+        @Override
+        public Iterator<Element> apply(final Iterator<Map.Entry<Key, Value>> entryIterator) {
+            final EntryToElement entryToElement = new EntryToElement(serialisedConf);
+            return entryIterator.map(entryToElement);
+        }
+    }
+
+    public static class EntryToElement extends AbstractFunction1<Map.Entry<Key, Value>, Element> implements Serializable {
+        private transient AccumuloElementConverter converter;
+        private transient View view;
+
+        public EntryToElement(final byte[] serialisedConf) {
+            try {
+                final Configuration conf = Utils.deserialiseConfiguration(serialisedConf);
+                final String keyPackageClass = conf.get(KEY_PACKAGE);
+                final Schema schema = Schema.fromJson(conf.get(SCHEMA).getBytes(CommonConstants.UTF_8));
+                final AccumuloKeyPackage keyPackage = Class
+                        .forName(keyPackageClass)
+                        .asSubclass(AccumuloKeyPackage.class)
+                        .newInstance();
+                keyPackage.setSchema(schema);
+                converter = keyPackage.getKeyConverter();
+                LOGGER.info("Initialised EntryToElement with AccumuloElementConverter of {}", converter.getClass().getName());
+                view = View.fromJson(conf.get(AbstractGetRDDHandler.VIEW).getBytes(CommonConstants.UTF_8));
+                LOGGER.info("Initialised EntryToElement with View of {}", view.toString());
+            } catch (final InstantiationException | IllegalAccessException | ClassNotFoundException | IOException e) {
+                throw new RuntimeException("Exception creating AccumuloKeyPackage from Configuration", e);
+            }
+        }
+
+        @Override
+        public Element apply(final Map.Entry<Key, Value> entry) {
+            final Element element = converter.getFullElement(entry.getKey(), entry.getValue(), false);
+            final ViewElementDefinition viewDef = view.getElement(element.getGroup());
+            if (viewDef != null) {
+                final ElementTransformer transformer = viewDef.getTransformer();
+                if (transformer != null) {
+                    transformer.apply(element);
```

Comment:
This doesn't look like it applies post transform filtering? Can we added that in?

Reply:
Added in. This was also not implemented in the `ElementInputFormat` approach. Fixed in both and a test added.